custom_components/spook/ectoplasms/cloud/switch.py

```diff
@@ -22,21 +23,15 @@
     from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 
-@dataclass(kw_only=True)
+@dataclass(frozen=True, kw_only=True)
 class HomeAssistantCloudSpookSwitchEntityDescription(
     SpookEntityDescription,
     SwitchEntityDescription,
 ):
     """Class describing Spook Home Assistant sensor entities."""
 
-    icon_off: str | None = None
-    is_on_fn: Callable[[Cloud], bool | None]
-    set_fn: Callable[[Cloud, bool], Awaitable[Any]]
-
-    def __post_init__(self) -> None:
-        """Sync icon_off with icon."""
-        if self.icon_off is None:
-            self.icon_off = self.icon
```

Comment:
This doesn't work for `frozen=True`. As it's always `None` always set to `self.icon`, I removed it.